posts/authentication-basic-cakephp-4.md

```diff
@@ -0,0 +1,154 @@
+---
+title: "Authentication basic CakePHP 4"
+date: "2020-11-06"
+published: true
+tags:
+  - php
+  - cakephp
+---
+
+Chào mọi người, hôm nay chúng ta hãy cùng nhau tìm hiểu về Authentication 2.0 trong Framework CakePHP 4 nhé!
+#I. Cài đặt Authentication 2.0
+Cài đặt plugin với composer từ thư mục ROOT của Dự án CakePHP của bạn (nơi chứa tệp **composer.json** ).
+
+`composer require cakephp/authentication:^2.0`
+
+Tải plugin authentication bằng cách thêm câu lệnh sau vào dự án của bạn trong `src/Application.php`:
+
+    public function bootstrap(): void
+    {
+        parent::bootstrap();
+        $this->addPlugin('Authentication');
+    }
+
+#II. Cấu hình và sử dụng Authentication để xác thực người dùng
+Plugin authentication được tích hợp trong dự án với vai trò như một middleware hay một component để giúp việc xác thực người dùng dễ dàng hơn. Đầu tiên, áp dụng nó như một middleware, thêm vào `src/Application.php` một số lớp cần thiết:
+ > 
+        use Authentication\AuthenticationService;
+        use Authentication\AuthenticationServiceInterface;
+        use Authentication\AuthenticationServiceProviderInterface;
+        use Authentication\Identifier\IdentifierInterface;
+        use Authentication\Middleware\AuthenticationMiddleware;
+        use Cake\Http\MiddlewareQueue;
+        use Psr\Http\Message\ServerRequestInterface;
+
+Tiếp theo, thêm `AuthenticationProviderInterface`vào lớp `Application`:
```

Comment:
dùng từ "thêm" không đúng lắm, ở đây AuthenticationServiceProviderInterface sẽ là interface cho class Application
extends là kế thừa toàn bộ, interface là kế thừa hình thức. e thử nghiên cứu xem dùng từ ngữ sao cho đúng nhé
hoặc e có thể dùng nguyên văn "implements" vào cx đc đây là góp ý của anh 😅